samples/nrf9160/modem_shell/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+#configuration for secure side
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/spm.conf")
+  set(spm_CONF_FILE
+    prj.conf
+    ${CMAKE_CURRENT_LIST_DIR}/spm.conf
+  )
+endif()
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(MOSH)
+target_include_directories(app PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
+  ${CMAKE_CURRENT_SOURCE_DIR}/iperf
+  ${CMAKE_CURRENT_SOURCE_DIR}/ping
+  ${CMAKE_CURRENT_SOURCE_DIR}/utils)
+
+if (DEFINED ENV{BUILD_ID})
+  zephyr_compile_definitions(
+    BUILD_ID=$ENV{BUILD_ID}
```

Comment:
What is a BUILD_ID, what is it used for, and why is it important ?

Why does this need to be specified as an environment variable, and not through Kconfig or a regualer CMake variable, like `-DBUILD_ID`.
I don't find any documentation on this.

Reply:
FyI Zephyr already provides `BUILD_VERSION` which can be defined by the user:
https://github.com/zephyrproject-rtos/zephyr/blob/main/CMakeLists.txt#L107-L109

BUILD_VERSION will default to `git describe` if not provided by the user.

Reply:
This comes from Jenkins environment and is set by the build system. I guess we could pass this via `-DBUILD_ID`.

Reply:
So because of Jenkins server then you add two environment undocumented environment variables to the build ?

Why is this specifically needed for mosh, when we don't need something similar for other samples ?
Remember, Zephyr already print build version:
```
*** Booting Zephyr OS build zephyr-v2.5.0-4098-ge2501fd08ef3  ***
```

Reply:
Other samples are not built in Jenkins so that they would be stored and shared from there to several stakeholders. Modem shell has Jenkins builds and we share those to different stakeholders (both internal and external) who perform testing.

Zephyr's version information doesn't tell anything about our application version and compile variant.

`-DBUILD_ID` would be ideal as that would only be visible in our Jenkins environment as well as the code that reads this parameter. However, when I tried this originally, it gives me this warning if I don't have anything defined in CMakeLists:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_ID
```
Would be good if you know how to get this working.

Reply:
@tejlmand This is something I'd like feedback on what would be correct approach in passing a BUILD_ID from compile flags or environment variable to the sample code.

Reply:
As just discussed, I'm removing `BUILD_ID `and `BUILD_VARIANT `from `CMakeLists.txt` and use `CONFIG_COMPILER_OPT `CMake option:
`west build -b nrf9160dk_nrf9160ns -d build -p -- -DCONFIG_COMPILER_OPT="\"-DBUILD_ID=123testing -DBUILD_VARIANT=something\""`